packages/block-library/src/navigation/edit/use-convert-classic-menu-to-block-menu.js

```
@@ -0,0 +1,122 @@
+	const [ error, setError ] = useState( null );
+
+	async function convertClassicMenuToBlockMenu( menuId, menuName ) {
+		let navigationMenu;
+		let classicMenuItems;
+
+		const menuItemsParameters = {
+			menus: menuId,
+			per_page: -1,
+			context: 'view',
+		};
+
+		const fetchError = new Error(
+			sprintf(
+				// translators: %s: the name of a menu (e.g. Header navigation).
+				__( `Unable to fetch classic menu "%s" from API.` ),
+				menuName
+			),
+			{
+				menuId,
+				menuName,
+			}
+		);
+
+		// 1. Fetch the classic Menu items.
+		try {
+			classicMenuItems = await registry
+				.resolveSelect( coreStore )
+				.getMenuItems( menuItemsParameters );
+		} catch ( e ) {
+			throw fetchError;
+		}
+
+		// Handle offline response which resolves to `null`.
+		if ( classicMenuItems === null ) {
+			throw fetchError;
+		}
+
+		// 2. Convert the classic items into blocks.
+		const { innerBlocks } = menuItemsToBlocks( classicMenuItems );
+
+		// 3. Create the `wp_navigation` Post with the blocks.
+		try {
+			navigationMenu = await createNavigationMenu(
+				menuName,
+				innerBlocks
+			);
+		} catch ( e ) {
+			throw new Error(
+				sprintf(
+					// translators: %s: the name of a menu (e.g. Header navigation).
+					__( `Unable to create Navigation Menu "%s".` ),
+					menuName
+				),
+				{
+					menuId,
+					menuName,
+				}
+			);
+		}
+
+		return navigationMenu;
+	}
+
+	const convert = useCallback(
+		( menuId, menuName ) => {
+			if ( ! menuId || ! menuName ) {
+				setError( 'Unable to convert menu. Missing menu details.' );
+				setStatus( CLASSIC_MENU_CONVERSION_ERROR );
+				return;
+			}
+
+			setStatus( CLASSIC_MENU_CONVERSION_PENDING );
+			setValue( null );
+			setError( null );
+
+			convertClassicMenuToBlockMenu( menuId, menuName )
+				.then( ( navMenu ) => {
+					setValue( navMenu );
+					setStatus( CLASSIC_MENU_CONVERSION_SUCCESS );
+				} )
+				.catch( ( e ) => {
+					setError( e?.message );
```

Let's re-throw this exception so that it can debugged in devtools.

---

Just checking...if we rethrow the error and it isn't handled then won't the block crash? I was trying to avoid that.

---

Ok I tested and it's all fine. The block handles the error but the message is still displayed in dev tools. Win 👍